scripts/09-history-migration-to-mongo.py

```diff
@@ -0,0 +1,455 @@
+        c1 = make_cast(self)
+        c2 = make_cast(other)
+        return c1 == c2
+
+    def dump(self):
+        return {
+            'set': [nbr.dump() for nbr in self.set],
+            'timestamp': self.timestamp,
+            'type': self.type,
+        }
+
+    def __repr__(self):
+        return '<GroupNodeBackendsSetRecord: set {}, timestamp {}, type {}>'.format(
+            [repr(nbr) for nbr in self.set],
+            self.timestamp,
+            self.type
+        )
+
+
+class GroupNodeBackendRecord(object):
+    def __init__(self, hostname, port, family, backend_id, path):
+        self.hostname = hostname
+        self.port = port
+        self.family = family
+        self.backend_id = backend_id
+        self.path = path
+
+    def __hash__(self):
+        return hash((self.hostname, self.port, self.family, self.backend_id, self.path))
+
+    def __eq__(self, other):
+        return (self.hostname == other.hostname and
+                self.port == other.port and
+                self.family == other.family and
+                self.backend_id == other.backend_id and
+                self.path == other.path)
+
+    def dump(self):
+        return {
+            'hostname': self.hostname,
+            'port': self.port,
+            'family': self.family,
+            'backend_id': self.backend_id,
+            'path': self.path,
+        }
+
+    def __repr__(self):
+        return (
+            '<GroupNodeBackendRecord: '
+            'hostname={hostname}, '
+            'port={port}, '
+            'family={family}, '
+            'backend_id={backend_id}, '
+            'path={path}>'.format(
+                hostname=self.hostname,
+                port=self.port,
+                family=self.family,
+                backend_id=self.backend_id,
+                path=self.path
+            )
+        )
+
+
+class GroupCoupleRecord(object):
+    # TODO: Move constats to history module
+    def __init__(self, couple, timestamp=None, type='automatic'):
+        self.couple = couple
+        self.timestamp = timestamp
+        self.type = type
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __eq__(self, other):
+        return set(self.couple) == set(other.couple)
+
+    def dump(self):
+        return {
+            'couple': self.couple,
+            'type': self.type,
+            'timestamp': self.timestamp,
+        }
+
+    def __repr__(self):
+        return '<GroupCoupleRecord: couple {}, timestamp {}, type {}>'.format(
+            self.couple,
+            self.timestamp,
+            self.type
+        )
+
+
+def migrate_history(s, coll):
+
+    ip_hostname_cache = {}
+
+    def ip_to_hostname(ip_address):
+        if ip_address in ip_hostname_cache:
+            return ip_hostname_cache[ip_address]
+
+        try:
+            res = socket.gethostbyaddr(ip_address)[0]
+        except Exception as e:
+            print 'Failed to resolve {}: {}'.format(ip_address, e)
```

Comment:
This required for some interaction with admin running the script. He rarely looks in log, only when something goes completely wrong. Otherwise, he should see the problems script is dealing with right away.